### ISSUE TITLE
Navigation block - Menu setting not preserved on "Save Changes"

### ISSUE THREAD

**Issue by jqmg** [WordPress/create-block-theme#574]
# Versions
WordPress 6.5.2
Twenty Twenty-Four 1.1
Create Block Theme 2.0.2

# Problem
I have a site header using two different menus. 
When I use the create block theme option "Save changes" both navigation blocks reference same menu.

Before "save changes" the menus in the code look like this:
```
<!-- wp:navigation {"ref":10} /-->
<!-- wp:navigation {"ref":4} /-->
```

After "save changes" the menus in the code look like this:
```
<!-- wp:navigation {"ref":10} /-->
<!-- wp:navigation {"ref":10} /-->
```

# How to replicate issue

1. Activate twentytwentyfour theme, install Create Block Theme.
2. Edit site / edit header / add two navigation blocks and create distinct menus for each.
3. Click Save, preview the site, everything looks fine here.
4. Now click the Create Block Theme wrench icon and click "Save Changes", both your navigation blocks are now using the same menu.

**Comment by madhusudhand** [comment]
Will be fixed in: #572

**Comment by pbking** [comment]
This should be fixed in #572 by leaving the "Remove Navigation Refs" option UNCHECKED when saving your site.

NOTE: That nav refs should still be removed when your theme is ready to distribute to another environment.

Closing this, but if you find it is still an issue please reopen and we will dig in further.
